src/main/asciidoc/index.adoc

```diff
@@ -17,503 +17,801 @@
+NOTE: Although often used interchangeably, the terms audit log and audit trail have different meanings.
+The term audit log refers to a chronological sequence of audit records (i.e., records that contain evidence related to
+and resulting from security-related events).
+The term audit trail refers to a set of audit records from one or more audit logs that relate to a specific event and/or
+user.
+
+What's more, the GDPR has made it a legal requirement when processing personally identifiable information.
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Logs MUST be written before and after all security-related events.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Logged data MUST be properly sanitized to prevent log injection attacks.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+If logged, tokens, credentials, personally identifiable information, or other sensitive information MUST be masked,
+sanitized, hashed, or encrypted.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Log integrity MUST be guaranteed at all times.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Logs MUST NOT be discarded before the end of the data retention period set by regulatory and contractual obligations,
+and MUST NOT be retained beyond that period.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+It is RECOMMENDED to use a separate partition for file-based logs that are persisted on the file system.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+If possible, strict file-level permissions SHOULD also be applied.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+It is RECOMMENDED to use a separate database account for logs that are persisted in a database.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+It's strongly RECOMMENDED to transmit it from distributed systems to a secure centralized log management system.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Although audit technical logs are typically collected for different purposes and kept separately, the two SHOULD have a
+consistent format to be easily correlated.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+It's RECOMMENDED to synchronize time across nodes to ensure consistent timestamps in logs.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Log monitoring and alerting MUST be automated whenever possible.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Alerts for potentially malicious activity MUST be manually reviewed by a human.
+====
+
+== Miscellaneous
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+When it comes to identifiers, sequential or otherwise predictable identifiers MUST NOT be used. Instead, it is strongly
+RECOMMENDED to use long, unique, random, and unpredictable identifiers such as
+https://datatracker.ietf.org/doc/html/rfc4122[RFC 4122]-defined UUIDs, also known as GUIDs.
+====
```

Comment:
to align with REST styleguide https://github.com/belgif/rest-guide/pull/84/files

Reply:
We'll have to align with identifiers-specific rules from the core G-Cloud REST guide.